pkg/controller/scheduledscaling/scaler.go

```diff
@@ -0,0 +1,118 @@
+package scheduledscaling
+
+import (
+	"fmt"
+	"time"
+
+	"golang.org/x/net/context"
+	autoscalingv1 "k8s.io/api/autoscaling/v1"
+	autoscalingv2 "k8s.io/api/autoscaling/v2"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	cacheddiscovery "k8s.io/client-go/discovery/cached/memory"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+	"k8s.io/client-go/scale"
+	scaleclient "k8s.io/client-go/scale"
+)
+
+// TargetScaler is an interface for scaling a target referenced resource in an
+// HPA to the desired replicas.
+type TargetScaler interface {
+	Scale(ctx context.Context, hpa *autoscalingv2.HorizontalPodAutoscaler, replicas int32) error
+}
+
+type hpaTargetScaler struct {
+	mapper          apimeta.RESTMapper
+	scaleNamespacer scaleclient.ScalesGetter
```

Comment:
The name `scaleNamespacer` is a bit confusing for me. From the code it seems to be a "typed" scale getter? Is there a way to name the variable that makes it more readable?